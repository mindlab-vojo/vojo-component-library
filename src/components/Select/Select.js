import React from 'react'
import PropTypes from 'prop-types'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import WindowedSelect from 'react-windowed-select'

const SelectMenuComponent = ({
  options,
  placeholder,
  onSelectChange,
  name,
  defaultInputValue
}) => {

  const styles = (theme) => {
    return {
      control: (styles, { isSelected, isFocused }) => ({
        ...styles,
        borderColor: isFocused ? theme.colors.primaryColor : isSelected ? theme.colors.primaryColor : 'white',
        boxShadow: '0px 5px 10px rgba(101, 101, 101, 0.05), 0px 5px 10px rgba(61, 61, 61, 0.1)',
        fontFamily: "'Rubik',sans-serif",
        fontWeight: '500',
        height: '48px',
        borderRadius: '4px',
        ':hover': {
          borderColor: isFocused ? theme.colors.primaryColor : isSelected ? theme.colors.primaryColor : 'white',
        }
      }),
      menu: (styles) => ({
        ...styles,
        fontFamily: "'Rubik',sans-serif",
        fontSize: '14px',
        color: '#4D4771'
      }),
      dropdownIndicator: (styles, { isSelected, isFocused }) => ({
        ...styles,
        color: theme.colors.primaryColor,
        border: 'none',
        ':hover': {
          color: theme.colors.darkerPrimaryColor
        }
      }),
      singleValue: (styles) => ({
        ...styles,
        color: '#4D4771',
        fontFamily: "'Rubik',sans-serif",
      }),
      indicatorSeparator: (styles) => ({
        border: 'none'
      }),
      placeholder: (styles) => ({
        ...styles,
        color: '#212121',
        fontWeight: '500',
        fontSize: '14px',
        fontFamily: "'Rubik',sans-serif",
      })
    }
  }

  const renderComponent = (theme) => (
    <WindowedSelect
      options={options}
      defaultInputValue={defaultInputValue}
      onChange={onSelectChange}
      name={name}
      placeholder={placeholder}
      styles={styles(theme)} />
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

SelectMenuComponent.propTypes = {
  options: PropTypes.any,
  placeholder: PropTypes.string,
  onSelectChange: PropTypes.func,
  name: PropTypes.string,
  defaultInputValue: PropTypes.any,
}
export default SelectMenuComponent
