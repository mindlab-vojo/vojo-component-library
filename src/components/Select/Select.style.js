import styled, { keyframes } from 'styled-components'

const slideIn = keyframes`
  from {
    margin-left: -360px;
  }
  to: {
    margin-left: 0;
  }

`
const slideOut = keyframes`
  from {
    margin-left: 0;
  }
  to: {
    margin-left: -360px;
  }
`

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to: {
    opacity: 0.3;
  }

`
const fadeOut = keyframes`
  from {
    opacity: 0.3;
  }
  to: {
    opacity: 0;
  }
`

export const Wrapper = styled.div`
  height: 100vh;
  left: 0;
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 4000;
`

export const Background = styled.div`
  animation: ${props => props.show ? fadeIn : fadeOut } ${props => `${props.animationDuration}s`} linear forwards;
  background-color: #000000;
  height: 100vh;
  left: 0;
  opacity: 0.3;
  position: absolute;
  top: 0;
  width: 100%;
  z-index: 4001;
`

export const Menu = styled.div`
  animation: ${props => props.show ? slideIn : slideOut } ${props => `${props.animationDuration}s`} linear forwards;
  background-color: #FFFFFF;
  box-sizing: border-box;
  height: 100vh;
  left: 0;
  max-width: 360px;
  padding: 25px;
  position: absolute;
  overflow-y: scroll;
  top: 0;
  width: 100%;
  z-index: 4002;

  &::-webkit-scrollbar {
    display: none;
  };
`

export const Close = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-bottom: 30px;
  width: 100%;
`

export const CloseIcon = styled.div`
  cursor: pointer;
  width: fit-content;
`

export const MenuItems = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

export const Item = styled.div`
  align-items: center;
  cursor: pointer;
  display: flex;
  margin: 7px 0px;
  width: 100%;

  &:first-of-type {
    margin-top: 0;
  };

  &:last-of-type {
    margin-bottom: 0;
  };
`

export const ItemIcon = styled.div`
  display: flex;
  width: 22px;
`

export const ItemText = styled.div`
  display: flex;
  margin-left: 15px;
  width: 100%;
`

export const MenuButtons = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 50px;
  width: 100%;
`

export const Button = styled.div`
  align-items: center;
  cursor: pointer;
  display: flex;
  margin: 7px 0px;
  width: 100%;

  &:first-of-type {
    margin-top: 0;
  };

  &:last-of-type {
    margin-bottom: 0;
  };
`

export const ButtonWrapper = styled.div`
  width: 128px;
`

export const Terms = styled.div`
  align-items: center;
  cursor: pointer;
  box-sizing: border-box;
  bottom: 20px;
  display: flex;
  justify-content: center;
  flex-direction: column;
  position: absolute;
`