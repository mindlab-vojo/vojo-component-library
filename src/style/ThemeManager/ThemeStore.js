import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore