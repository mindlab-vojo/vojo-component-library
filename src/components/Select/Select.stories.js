import React from 'react';
import { withKnobs, text, select, boolean } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import Select from '.';
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

const options = [{
  value: 1,
  description: "Sem ",
  code: "NDA",
},
{
  value: 2,
  description: "Ensino",
  code: "EFI",
}, 
{
  value: 3,
  description: "Ensino",
  code: "EFC",
}, 
{
  value: 4,
  description: "Ensino",
  code: "EMI",
},]
const list = {
  Filled: 'Filled',
  Outlined: 'Outlined',
  Standard: 'Standard'
}
const inputLabel = text('inputLabel', 'Grau')
const variant = select('variant',  list)

export const selectStorie = () => {
  return (
    <Select 
      options={options}
      inputLabel={inputLabel} 
      value={'value'}
      variant={variant}/>
  )
}

export default {
  title: categories.Selects,
  name: 'Select',
  component: selectStorie,
  decorators: [
    withKnobs,
    withA11y,
    (story) => <ThemeProvider>{story()}</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{story()}</div>,
  ],
}
