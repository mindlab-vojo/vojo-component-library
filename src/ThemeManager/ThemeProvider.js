import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer