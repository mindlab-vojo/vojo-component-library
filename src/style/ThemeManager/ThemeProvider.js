import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer