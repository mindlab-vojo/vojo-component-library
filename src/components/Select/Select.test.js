import React from 'react';
import { render } from '@testing-library/react';

import Select from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test Select component render', () => {
  const fakeFn = jest.fn()
  const component = render(
    <ThemeProvider>
      <Select 
        options={[{
          value: 1,
          description: "Sem instrução",
          code: "NDA",
        },
        {
          value: 2,
          description: "Ensino fundamental incompleto ou equivalente",
          code: "EFI",
        }, 
        {
          value: 3,
          description: "Ensino fundamental completo ou equivalente",
          code: "EFC",
        }, 
        {
          value: 4,
          description: "Ensino médio incompleto ou equivalente",
          code: "EMI",
        },]}/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
