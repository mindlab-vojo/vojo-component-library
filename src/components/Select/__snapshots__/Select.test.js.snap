// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root makeStyles-formControl-1"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        />
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        >
          <select
            aria-invalid="false"
            class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
          >
            <option
              aria-label="None"
              value=""
            />
            <option
              value="NDA"
            >
              Sem instrução
            </option>
            <option
              value="EFI"
            >
              Ensino fundamental incompleto ou equivalente
            </option>
            <option
              value="EFC"
            >
              Ensino fundamental completo ou equivalente
            </option>
            <option
              value="EMI"
            >
              Ensino médio incompleto ou equivalente
            </option>
          </select>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink="false"
      />
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <select
          aria-invalid="false"
          class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
        >
          <option
            aria-label="None"
            value=""
          />
          <option
            value="NDA"
          >
            Sem instrução
          </option>
          <option
            value="EFI"
          >
            Ensino fundamental incompleto ou equivalente
          </option>
          <option
            value="EFC"
          >
            Ensino fundamental completo ou equivalente
          </option>
          <option
            value="EMI"
          >
            Ensino médio incompleto ou equivalente
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
